bika/lims/exportimport/instruments/resultsimport.py

```diff
@@ -987,3 +990,18 @@ def _process_analysis(self, objid, analysis, values):
         if len(fields_to_reindex):
             analysis.reindexObject(idxs=fields_to_reindex)
         return resultsaved
+
+    def toAutoSubmitInterims(self, analysis, interimsout=[]):
+        """Returns True or False depending of wheather there are missing interims fields
+           and if those missing interims fields are used in a calculation.
+        """
+        if not interimsout:
+            return True
+        interims_with_no_values = filter(lambda interim: not interim.get('value', False), interimsout)
+        if interims_with_no_values:
+            calculation = analysis.getCalculation()
+            formula = calculation.getMinifiedFormula()
+            missing_vals_in_formula = filter(lambda interim: (interim.get('keyword') in formula), interims_with_no_values)
+            if missing_vals_in_formula:
+                return False
+        return True
```

Comment:
@xispa I've made the changes you have asked for but I did push them because the test fails
This line if `not interim.get("value", None)` fails when value is zero and this value is one `calc.getInterimFields` and therefore the analysis does not get submitted.